internal/pkg/repository/repository.go

```diff
@@ -45,35 +49,24 @@ func NewWithURI(registry Registry, name, uri string) *Repository {
 		name:     name,
 		registry: registry,
 		uri:      uri,
+		docker:   dockerengine.New(exec.NewCmd()),
 	}
 }
 
 // BuildAndPush builds the image from Dockerfile and pushes it to the repository with tags.
-func (r *Repository) BuildAndPush(docker ContainerLoginBuildPusher, args *dockerengine.BuildArguments) (digest string, err error) {
+func (r *Repository) BuildAndPush(args *dockerengine.BuildArguments) (digest string, err error) {
 	if args.URI == "" {
 		uri, err := r.URI()
 		if err != nil {
 			return "", err
 		}
 		args.URI = uri
 	}
```

Comment:
do we still need this chunk?

Reply:
This can be removed. we can just use `args.URI` in `workload`

Reply:
I removed the logic to set URI in BuildArgs for `workload/task_run`. Then we need to need the above chunk.